assets/app/actions/siteActions.js

```diff
@@ -77,21 +77,35 @@ export default {
     }).catch(error => alertActions.alertError(error.message));
   },
 
+  createPR(site, head, base) {
+    return github.createPullRequest(site, head, base).then((pr) => {
+      return github.mergePullRequest(site, pr);
+    }).then(() => {
+      return github.deleteBranch(site, head);
+    }).then(() => {
+      this.fetchBranches(site);
+    }).then(() => {
+      return alertActions.alertSuccess(`${head} merged successfully`);
+    }).catch(error => alertActions.httpError(error.message));
```

Comment:
I think if the delete branch failed we would want to have some mechanism to clean it later, although I'm not sure if that scenario is going to be common enough to warrant a change at this point. 

I do with you could make transactional calls to the github api, and they could just roll back your request if one of the operations failed.